### ISSUE TITLE
Tag Based "Stay in Contact" Reminders

### ISSUE THREAD

**Issue by chrisspiegl** [monicahq/monica#1940]
I am starting to migrate more and more data into Monica to stay up to date with my relationships (personal and business).

I also recently read an article on how to better stay in contact with loads of people.

But what I noticed is that Monica is very "1-on-1". Meaning that I have to do everything for everybody in a single action.

One main idea I had was to make it possible to have "Stay in Contact" reminders based on tags. This would give the ability to tag people as "Family" and contact them once per week but "Business Level 1" only every 3 weeks, and "Business Level 3" only every 6 months.

I think this could be hugely beneficial to speed up the management of this.

On top of easier management, these reminders could also be sent in one email rather than however many contacts there are.

Looking forward to feedback on this idea and to potentially see it in Monica soon 💪.

**Comment by lydgate** [comment]
Was it Derek Sivers [article](http://sivers.org/hundreds) you read, out of curiosity? :)

I implemented this in SQL. Basically I constructed a view that did a UNION across calls, conversations, notes, and activities, to get the timestamps from each table into rows matched with a user id/name, and limit it to the max timestamp by user. Then I query that view using a combination of tags and the is_active column to get the behaviour you describe.

This lets me have two tags that correspond to "contact every 3 weeks" and "every 2 months". Then default contacts should be "every 6 months", whereas archived contacts have this feature disabled. 

I'm experimenting with how to alert; at the moment I have a cached version of the alert show up at the start of my shell prompt, but I've also hooked it into Todoist. I may get rid of one method at some point.

**Comment by chrisspiegl** [comment]
Yes, it is Derek Sivers that inspired me to ask this question. Right now I am still very bad at even keeping my connections organized. The ability to easily input is not really there yet and the overhead is crazy for my life.

But I can absolutely see the benefits and would appreciate an ability to have this kind of "connect with this week" kind of features in the Monica interface to just batch process these every week on Mondays or something similar.

I am mostly interested if it's integrated though and not a separate step once again.
